improver_tests/calibration/reliability_calibration/test_ManipulateReliabilityTable.py

```diff
@@ -105,6 +105,261 @@ def setUp(self):
+            obs_count,
+            forecast_probability_sum,
+            forecast_count,
+            self.probability_bin_coord,
+        )
+
+        assert_array_equal(result[:3], expected)
+        expected_bin_coord_points = np.array([0.1, 0.3, 0.6, 0.9], dtype=np.float32)
+        expected_bin_coord_bounds = np.array(
+            [[0.0, 0.2], [0.2, 0.4], [0.4, 0.8], [0.8, 1.0]], dtype=np.float32,
+        )
+        assert_allclose(expected_bin_coord_points, result[3].points)
+        assert_allclose(expected_bin_coord_bounds, result[3].bounds)
+
+    def test_two_undersampled_bins(self):
+        """Test when two bins are under-sampled."""
+        obs_count = np.array([0, 12, 250, 75, 250], dtype=np.float32)
+        forecast_probability_sum = np.array([0, 12, 250, 75, 250], dtype=np.float32)
+        forecast_count = np.array([1000, 50, 500, 100, 250], dtype=np.float32)
+
+        expected = np.array(
+            [
+                [0, 262, 325],  # Observation count
+                [0, 262, 325],  # Sum of forecast probability
+                [1000, 550, 350],  # Forecast count
+            ]
+        )
+        result = self.plugin._combine_undersampled_bins(
+            obs_count,
+            forecast_probability_sum,
+            forecast_count,
+            self.probability_bin_coord,
+        )
+
+        assert_array_equal(result[:3], expected)
+        expected_bin_coord_points = np.array([0.1, 0.4, 0.8], dtype=np.float32)
+        expected_bin_coord_bounds = np.array(
+            [[0.0, 0.2], [0.2, 0.6], [0.6, 1.0]], dtype=np.float32,
+        )
+        assert_allclose(expected_bin_coord_points, result[3].points)
+        assert_allclose(expected_bin_coord_bounds, result[3].bounds)
+
+    def test_two_equal_undersampled_bins(self):
+        """Test when two bins are under-sampled and the under-sampled bins have
+        an equal forecast count."""
+        obs_count = np.array([0, 25, 250, 75, 250], dtype=np.float32)
+        forecast_probability_sum = np.array([0, 25, 250, 75, 250], dtype=np.float32)
+        forecast_count = np.array([1000, 100, 500, 100, 250], dtype=np.float32)
+
+        expected = np.array(
+            [
+                [0, 275, 325],  # Observation count
+                [0, 275, 325],  # Sum of forecast probability
+                [1000, 600, 350],  # Forecast count
+            ]
+        )
+
+        result = self.plugin._combine_undersampled_bins(
+            obs_count,
+            forecast_probability_sum,
+            forecast_count,
+            self.probability_bin_coord,
+        )
+
+        assert_array_equal(result[:3], expected)
+        expected_bin_coord_points = np.array([0.1, 0.4, 0.8], dtype=np.float32)
+        expected_bin_coord_bounds = np.array(
+            [[0.0, 0.2], [0.2, 0.6], [0.6, 1.0]], dtype=np.float32,
+        )
+        assert_allclose(expected_bin_coord_points, result[3].points)
+        assert_allclose(expected_bin_coord_bounds, result[3].bounds)
+
```

Comment:
The only thing that leaps out at me is that we haven't demonstrated how multiple undersampled neighbours get stacked up, e.g.
```
    def test_three_equal_undersampled_bin_neighbours(self):
        obs_count = np.array([0, 25, 50, 75, 250], dtype=np.float32)
        forecast_probability_sum = np.array([0, 25, 50, 75, 250], dtype=np.float32)
        forecast_count = np.array([1000, 100, 100, 100, 250], dtype=np.float32)

        expected = np.array(
            [
                [0, 150, 250],  # Observation count
                [0, 150, 250],  # Sum of forecast probability
                [1000, 300, 250],  # Forecast count
            ]
        )
```

Reply:
I've added this additional test.